website/addons/wiki/views.py

```diff
@@ -245,14 +219,45 @@ def project_wiki_delete(auth, wname, **kwargs):
 
 
 @must_be_valid_project  # returns project
-@must_have_permission('write')  # returns user, project
-@must_not_be_registration
+@must_be_contributor_or_public
 @must_have_addon('wiki', 'node')
-def project_wiki_edit(auth, wname, **kwargs):
+def project_wiki_view(auth, wname, path=None, **kwargs):
     node = kwargs['node'] or kwargs['project']
-    wiki_name = wname.strip()
+    anonymous = has_anonymous_link(node, auth)
+    wiki_name = (wname or '').strip()
     wiki_key = to_mongo_key(wiki_name)
     wiki_page = node.get_wiki_page(wiki_name)
+    toc = _serialize_wiki_toc(node, auth=auth)
+    can_edit = node.has_permission(auth.user, 'write') and not node.is_registration
+    versions = _get_wiki_versions(node, wiki_name, anonymous=anonymous)
+
+    # Parse URL to determine default panel display
+    view_settings = {}
+    param_list = path.lower().split('/') if path else []
```

Comment:
Yeah, I tend to agree. We probably want three main views (/view/, /edit/, and /compare/) and then allow changes to those base views with query parameters. The implication is that there would be a hierarchy of base views. For example, /edit/ would make the edit and view columns open, /view/ would be just the view column, and /compare/ would be view and compare. If any of those states are active, the urls would match that. Having all three would be edit with a ?with=compare (and maybe something about the menu state), or if you wanted edit and compare without view you might do ?with=compare&without=view. Alternately, we could have any deviation list out the totality of the column state so we don't have to do plus/minus style routes. 

What I'm saying is we should chat for a couple of minutes.